src/libs/OptionsListUtils.ts

```diff
@@ -199,58 +306,51 @@ function getPersonalDetailsForAccountIDs(accountIDs, personalDetails) {
 
 /**
  * Return true if personal details data is ready, i.e. report list options can be created.
- * @param {Object} personalDetails
- * @returns {Boolean}
  */
-function isPersonalDetailsReady(personalDetails) {
-    return !_.isEmpty(personalDetails) && _.some(_.keys(personalDetails), (key) => personalDetails[key].accountID);
+function isPersonalDetailsReady(personalDetails: OnyxEntry<PersonalDetailsList>): boolean {
+    const personalDetailsKeys = Object.keys(personalDetails ?? {});
+    return personalDetailsKeys.some((key) => personalDetails?.[Number(key)]?.accountID);
```

Comment:
Q: why do we need to convert `key` to number?

Reply:
Its not needed anymore , will remove it